### DISCUSSION TITLE
Capturing destination_addr in set_message_sent_handler

### DISCUSSION THREAD

**Discussion by tahseenjamal** [python-smpplib/python-smpplib#173] [Q&A] [answered]
Currently the handler is 

`client.set_message_sent_handler(lambda pdu: sys.stdout.write(f"{ pdu.message_id }\n"))`

This I can log also but I want to log with it, the `destination_addr` it corresponds to. How do I find `destination_addr` that is related to this `message_id` received on `submit_sm`? 

Basically I want to note the `message_id` and `destination_addr` in a single line

**Comment by eigenein** [comment, marked as answer]
Hi @tahseenjamal,

This is somewhat out of scope, since SMPP doesn't send any message data back in a `SubmitSM` response – just the message ID. You need to store the rest yourself, should you need it later.

**Comment by tahseenjamal** [comment]
@eigenein Thanks
I have switched to Golang go-smpp which has lot of features. Correct me if am wrong, this python smpp project is not actively developed anymore
